kupfer/plugin/pinboard.py

```diff
@@ -0,0 +1,42 @@
+__kupfer_name__ = _("Pinboard")
+__kupfer_sources__ = ("PinboardBookmarkSource",)
+__kupfer_actions__ = ()
+__description__ = _("Manage and use bookmarks from Pinboard")
+__version__ = ""
+__author__ = "Peter Stuifzand <peter@p83.nl>"
+
+from kupfer import plugin_support
+from kupfer.objects import Source
+from kupfer.objects import UrlLeaf
+import pinboard
+
+__kupfer_settings__ = plugin_support.PluginSettings(
+    {
+        "key": "token",
+        "label": _("Pinboard API Token"),
+        "type": str,
+        "value": "",
+    },
+)
+
+class PinboardBookmarkSource(Source):
+    def __init__(self):
+        super().__init__(_("Pinboard Bookmarks"))
+        token = __kupfer_settings__["token"]
+        self.pinboard = pinboard.Pinboard(token)
```

Comment:
Token in this place may be not initialized or empty (first run).
self.pineboard should be initialized at first get_items. Maybe even not should be object attribute  as long is only used in get_items

Reply:
I squashed the commits and look if token is set.